packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -325,7 +339,7 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
           if (torrent.paused) {
             // We currently treat pausing torrent as canceling downloads
             await this.closeDownloadingChannels(torrent);
-          } else if (!torrent.done) {
+          } else if (!torrent.done || !torrent.destroyed) {
```

Comment:
this fixes a race condition that came up a couple of times, where just after cancelling the download, the leecher tried to make a new payment (it threw an error because the channel was already closed by that point, but I wanted to avoid that)